agent/acs/handler/acs_handler.go

```diff
@@ -332,8 +335,13 @@ func (acsSession *session) startACSSession(client wsclient.ClientServer) error {
 
 	client.AddRequestHandler(payloadHandler.handlerFunc())
 
-	// Ignore heartbeat messages; anyMessageHandler gets 'em
-	client.AddRequestHandler(func(*ecsacs.HeartbeatMessage) {})
+	// Add HeartbeatHandler to acknowledge ACS heartbeats
+	heartbeatHandler := newHeartbeatHandler(acsSession.ctx, client)
+	defer heartbeatHandler.clearAcks()
```

Comment:
defer statements for a FILO stack in golang, so just wanted to verify here, do we want `.stop()` to be called _before_ `.clearAcks()`?

Secondly, there doesn't appear to be any use-case to clearAcks besides cleanup, so could we just integrate it into the stop() function to make this more fool-proof?

Reply:
For the order --- yes, it is intended for `.stop()` (context cancellation) to be invoked before `.clearAcks()` (channel draining) to ensure no message is left.

This behavior is mostly borrowed from `refreshCredsHandler`, `taskManifestHandler`, and `payloadHandler`. (https://github.com/aws/amazon-ecs-agent/blob/master/agent/acs/handler/acs_handler.go#L316-L331 ) I thought about refactoring it to be more concise but ended up keeping it as-is so we have consistent implementation across ACS handlers. Maybe in future we will have another layer of interface abstraction?

Let me know which route is preferred.

Reply:
okay, makes sense, since it's consistent just leave it as-is